be/src/pipeline/exec/partitioned_aggregation_source_operator.cpp

```diff
@@ -133,13 +133,16 @@ Status PartitionedAggSourceOperatorX::get_block(RuntimeState* state, vectorized:
     SCOPED_TIMER(local_state.exec_time_counter());
     RETURN_IF_ERROR(local_state._status);
 
-    RETURN_IF_ERROR(local_state.initiate_merge_spill_partition_agg_data(state));
+    if (local_state._shared_state->is_spilled) {
+        RETURN_IF_ERROR(local_state.initiate_merge_spill_partition_agg_data(state));
```

Comment:
why call init merge every time?? It is only called the first time or meet previous partition's EOS?